blog/content/post/2020-04-03-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,796 @@
+      initrd ($drive2)/nix/store/zv2vl35xldkbss1y2fib1nifmw0yvick-initrd-linux-5.1.0/initrd
+    }
+    (...)
+    ```
+
+With `grub.cfg` content as above `configuration.nix` must have
+`boot.loader.grub.extraEntries `line like this:
+
+    ```bash
+    $ cat /etc/nixos/configuration.nix
+      (...)
+      boot.loader.grub.extraEntries = ''
+      menuentry "NixOS - Secure Launch" {
+      search --set=drive1 --fs-uuid 178473b0-282f-4994-96fc-a8e51e2cfdac
+      search --set=drive2 --fs-uuid 178473b0-282f-4994-96fc-a8e51e2cfdac
+        slaunch skinit
+        slaunch_module ($drive2)/boot/lz_header
+        linux ($drive2)/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage systemConfig=/nix/store/b32wgz392q99cls12pkd8adddzbdkprn-nixos-system-nixos-20.09.git.50c3e448fceM init=/nix/store/b32wgz392q99cls12pkd8adddzbdkprn-nixos-system-nixos-20.09.git.50c3e448fceM/init console=ttyS0,115200 earlyprintk=serial,ttyS0,115200 loglevel=4
+        initrd ($drive2)/nix/store/zv2vl35xldkbss1y2fib1nifmw0yvick-initrd-linux-5.1.0/initrd
+      }
+    '';
+    ```
+
+If there are differences in any of `search --set=drive1...`, `search
+--set=drive2...`, `linux ($drive2)/nix/store...` lines. Edit
+`configuration.nix` content and copy those lines from `grub.cfg`.
+
+16. Update system again.
+
+    ```bash
+    $ sudo nixos-rebuild switch -I nixpkgs=~/nixpkgs
+    ```
+
+17. Reboot platform.
+
+During platform booting, in GRUB menu there should be at least `"NixOS -
+Default"` and `"NixOS - Secure Launch"` entries. First entry boots platform
+without DRTM. Second entry **executes DRTM**! Choose the second entry and see if
+platform boots successfully. If yes, you have secure platform with DRTM enabled.
+
+## Validation
+
+You can still be suspicious, if it really works. And rightly so. In this
+section, I will show you, how to verify each component of the system to make you
+sure about its correctness. Also, I will present how we met first stage
+project's requirement.
+
+### GRUB
+
+There are two ways to validate if GRUB will load `slaunch` module and hence run
+SKINIT and LZ (DRTM).
+
+##### Verify content of `grub.cfg` file.
+
+    ```
+    $ cat /boot/grub/grub.cfg
+    menuentry "NixOS - Secure Launch" {
+    search --set=drive1 --fs-uuid babbf771-2766-4f6e-9a1f-7752464a6d37
+    search --set=drive2 --fs-uuid babbf771-2766-4f6e-9a1f-7752464a6d37
+      slaunch skinit
+      slaunch_module ($drive2)/boot/lz_header
+      linux ($drive2)/nix/store/ymvcgas7b1bv76n35r19g4p142v4cr0b-linux-5.1.0/bzImage systemConfig=/nix/store/j6l74m01hh5hkargzdbwzhpsdhgzwsgp-nix4
+      initrd ($drive2)/nix/store/n28axzys9kh4d4rvnpxv38pparzmkcfi-initrd-linux-5.1.0/initrd
+
+    }
+    ```
+
+There must be `slaunch skinit` entry and `slaunch_module
+($drive2)/boot/lz_header` which points to LZ.
+
+##### Compare bootlog with DRTM and without DRTM
+
+1. Reboot platform. In GRUB menu choose `"NixOS - Default"` entry. Verify logs.
```

Comment:
Please explain `Verify logs` here. Pre-kernel lines won't be visible in `dmesg`, and that's the first thing I would use to check the logs.